apiserver/pkg/storage/calico/resource.go

```diff
@@ -428,15 +427,16 @@ func (rs *resourceStore) GuaranteedUpdate(
 		if err != nil {
 			return err
 		}
-		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
+		versionStr := accessor.GetResourceVersion()
+		var revInt uint64
+		if versionStr != "" {
+			revInt, err = strconv.ParseUint(versionStr, 10, 64)
```

Comment:
I think we need to check if the err is non-nil here

Reply:
Oops, yep, reworked the code and left it half baked